apis/r/R/SOMASparseNDArrayRead.R

```diff
@@ -7,51 +117,153 @@
+    #' Matrix::\link[Matrix]{dgTMatrix-class}.
+    #'
     #' @return \link{SparseReadIter}
+    #'
     sparse_matrix = function(zero_based=FALSE) {
       #TODO implement zero_based argument, currently doesn't do anything
 
-
-      if (any(private$shape > .Machine$integer.max)) {
+        shape <- self$shape
+      # if (any(private$shape > .Machine$integer.max)) {
+      if (any(shape > .Machine$integer.max)) {
         warning(
           "Array's shape exceeds '.Machine$integer.max'.\n",
           "  - Result will only include coordinates within [0, 2^31 - 1).\n",
           "  - The full range of coordinates can be obtained with $tables().",
           call. = FALSE,
           immediate. = TRUE
         )
-        private$shape <- pmin(private$shape, .Machine$integer.max)
+        # private$shape <- pmin(private$shape, .Machine$integer.max)
+        shape <- pmin(shape, .Machine$integer.max)
       }
 
-      SparseReadIter$new(private$sr, private$shape, zero_based = zero_based)
+      SparseReadIter$new(self$sr, shape, zero_based = zero_based)
     },
 
     #' @description Read as a arrow::\link[arrow]{Table} (lifecycle: experimental).
     #' Returns an iterator of arrow::\link[arrow]{Table}.
+    #'
     #' @return \link{TableReadIter}
+    #'
     tables = function() {
-      TableReadIter$new(private$sr)
+      TableReadIter$new(self$sr)
+    },
+    #' @description Read in a blockwise fashion
+    #'
+    #' @template param-blockwise-iter
+    #' @template param-dots-ignored
+    #'
+    #' @return A \code{\link{SOMASparseNDArrayBlockwiseRead}} iterated reader
+    #'
+    blockwise = function(
+      axis,
+      ...,
+      size = NULL,
+      reindex_disable_on_axis = NULL
+    ) {
+      return(SOMASparseNDArrayBlockwiseRead$new(
+        self$sr,
+        self$array,
+        self$coords,
+        axis,
+        size = size,
+        reindex_disable_on_axis = reindex_disable_on_axis
+      ))
     }
-  ),
+  )
+)
 
+#' Blockwise Sparse ND-Array Reader
+#'
+#' @description Blockwise reader for \code{\link{SOMASparseNDArray}}
+#'
+#' @keywords internal
+#'
+#' @export
+#'
+SOMASparseNDArrayBlockwiseRead <- R6::R6Class(
+  classname = "SOMASparseNDArrayBlockwiseRead",
+  inherit = SOMASparseNDArrayReadBase,
+  cloneable = FALSE,
+  public = list(
+    #' @description Create
+    #'
+    #' @template param-blockwise-iter
+    #' @template param-coords-read
+    #' @template param-dots-ignored
+    #'
+    initialize = function(
+      sr,
+      array,
+      coords,
+      axis,
+      ...,
+      size,
+      reindex_disable_on_axis = NULL
+    ) {
+      super$initialize(sr, array, coords)
+      stopifnot(
+        "'size' must be a single integer value" = is.null(size) ||
+          rlang::is_integerish(size, 1L, finite = TRUE) ||
+          (inherits(size, 'integer64') && length(size) == 1L && is.finite(size)),
+        "'reindex_disable_on_axis' must be  avector of integers" = is.null(reindex_disable_on_axis) ||
```

Comment:
```suggestion
        "'reindex_disable_on_axis' must be a vector of integers" = is.null(reindex_disable_on_axis) ||
```

Reply:
Thanks @mlin, corrected in https://github.com/single-cell-data/TileDB-SOMA/pull/2152/commits/839411a3433cd181f91b6d1642562fd4cfcddfe2